### ISSUE TITLE
1.3.0 DecimalError bug

### ISSUE THREAD

**Issue by RickWong** [recharts/recharts#1493]
## Do you want to request a feature or report a bug?

Bug. 

## What is the current behavior?

Throws DecimalError.

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar (template: http://jsfiddle.net/ndLhnegs/).

```
[DecimalError] Invalid argument: Infinity at new Decimal (http://localhost:3000/static/js/bundle.js:14958:17) at getTickValuesFixedDomainFn 
```

## What is the expected behavior?

Version 1.3.0 is compatible with 1.2.0.

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?

1.3.0, bug found in latest Chrome.

**Comment by arcthur** [comment]
I am not find decimal error use this demo on local.

**Comment by johnsmith1st** [comment]
Also got this error when the data prop for LineChart is empty array or null. Works well in v1.2.0

**Comment by arcthur** [comment]
@johnsmith1st  empty array or null also no error, could you provide a jsfiddle demo

**Comment by wopian** [comment]
Haven't got the time to debug where/what its being called with until later this week, but I've also encountered this error after upgrading to 1.3.0.

Reproducible with https://github.com/wopian/kitsu-season-trends/tree/renovate/recharts-1.x

**Comment by arcthur** [comment]
I tyied react 15 or 16, and found no error above. please give me a demo.

**Comment by iamanoopmk** [comment]
Here is a fiddle that gives the error
https://jsfiddle.net/xqjtetw0/16558/

**Comment by arcthur** [comment]
I use this fiddle method to render chart, but no error.
![image](https://user-images.githubusercontent.com/322331/46419935-8f9d3580-c761-11e8-84fd-b47e834496f5.png)

**Comment by iamanoopmk** [comment]
Here is the error that pops up in console and the chart is not displayed for me. The fiddle link can be seen in the address bar.

![screenshot from 2018-10-03 20-42-20](https://user-images.githubusercontent.com/1889875/46420082-e4828100-c74c-11e8-9321-1c7a30f0c09f.png)

Here is the log message

Uncaught Error: Minified React error #109; visit http://facebook.github.io/react/docs/error-decoder.html?invariant=109&args[]=e for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at r (react-dom.min.js:13)
    at p._renderValidatedComponent (react-dom.min.js:11)
    at performInitialMount (react-dom.min.js:11)
    at p.mountComponent (react-dom.min.js:11)
    at Object.mountComponent (react-dom.min.js:12)
    at mountChildren (react-dom.min.js:12)
    at m._createInitialChildren (react-dom.min.js:11)
    at mountComponent (react-dom.min.js:11)
    at Object.mountComponent (react-dom.min.js:12)
    at performInitialMount (react-dom.min.js:11)

**Comment by arcthur** [comment]
because you use 1.3.0, that don't support react 15. I update 1.3.1 yesterday.
https://jsfiddle.net/xqjtetw0/16562/

**Comment by iamanoopmk** [comment]
@arcthur 
Yup. Sorry. My Mistake. I did update my libraries. The is gone now. Thank you.

**Comment by wopian** [comment]
~~Finally had the time to debug. This error shows up (in 1.3.0 and 1.3.1) when `AreaChart` receives an empty array.~~

I can reproduce this with `domain={['dataMin', 'dataMax']}` on an axis. This error occurs when data is an empty array and when the value of the `dataKey` used is not a Number (i.e a String) https://codepen.io/wopian/pen/vVLobr?editors=0010#0

This produces two errors:
- `Error: [DecimalError] Invalid argument: Infinity`
- `Uncaught Error: [DecimalError] Invalid argument: Infinity`

```js
const { AreaChart, XAxis, YAxis, Area } = Recharts

const data = [
  /* This will work fine
  { name: 1, pv: 2400 },
  { name: 2, pv: 1398 }
  */
  /* [DecimalError] Invalid argument: Infinity
  { name: "1", pv: 2400 },
  { name: "2", pv: 1398 }
  */
]

class App extends React.Component {
  render() {
    return (
      <AreaChart width={600} height={300} data={data} >
        <XAxis dataKey="name" type='number' domain={['dataMin', 'dataMax']}/>
        <YAxis/>
        <Area dataKey="pv" />
      </AreaChart>
    )
  }
}

ReactDOM.render(<App />, document.getElementById('container'))
```

**Comment by arcthur** [comment]
I fixed by 1.3.2

**Comment by wopian** [comment]
`1.3.2` still produces the same React-crashing error.

Using 1.3.2: https://codepen.io/wopian/pen/vVLobr?editors=0010#0

**Comment by xile611** [comment]
@wopian It's ok to use 1.3.3
https://codepen.io/anon/pen/bmWNVb?editors=0010

**Comment by laura-brouckman** [comment]
After upgrading to recharts `1.6.x` I am seeing this issue on a basic line chart with `data = []` and the domain set to be `["dataMin", "dataMax"]`

**Comment by artin-phares** [comment]
I'm receiving "[DecimalError] Invalid argument: NaN" in `1.6.0` with this:  

`<XAxis dataKey="time" type="number" domain={['dataMin', 'dataMax']} />`
and empty data array.

https://codepen.io/anon/pen/xNdZgQ?editors=0010

**Comment by TheTedAdams** [comment]
This seems to have been resolved in `1.6.1` for me.
